tests/test_worker.py

```diff
@@ -2622,6 +2622,49 @@ def mock_stream(*args, **kwargs):
 
         assert result.status_code == -1
 
+    async def test_watch_handles_410(
+        self,
+        default_configuration: KubernetesWorkerJobConfiguration,
+        flow_run,
+        mock_batch_client,
+        mock_core_client,
+        mock_watch,
+    ):
+        mock_watch.stream.side_effect = [
+            _mock_pods_stream_that_returns_running_pod(),
+            _mock_pods_stream_that_returns_running_pod(),
+            ApiException(status=410),
+            _mock_pods_stream_that_returns_running_pod(),
+        ]
+
+        # The job should not be completed to start
+        mock_batch_client.read_namespaced_job.return_value.status.completion_time = None
+
+        async with KubernetesWorker(work_pool_name="test") as k8s_worker:
+            await k8s_worker.run(flow_run=flow_run, configuration=default_configuration)
+
+        mock_watch.stream.assert_has_calls(
+            [
+                mock.call(
+                    func=mock_core_client.list_namespaced_pod,
+                    namespace=mock.ANY,
+                    label_selector=mock.ANY,
+                    timeout_seconds=60,
+                ),
+                mock.call(
+                    func=mock_batch_client.list_namespaced_job,
+                    namespace=mock.ANY,
+                    field_selector=mock.ANY,
+                ),
+                mock.call(
+                    func=mock_batch_client.list_namespaced_job,
+                    namespace=mock.ANY,
+                    field_selector=mock.ANY,
+                    resource_version=mock.ANY,
```

Comment:
what do you think about checking for a specific value of `resource_version` here?

Reply:
the only reason I didn't is because I'm not sure how I'd do it. I think I'd need to mock `list_namespaced_job` to return it for a specific call, right?